src/components/CornerDialog/CornerDialog.props.ts

```diff
@@ -0,0 +1,13 @@
+import type { ReactNode } from 'react';
+
+import { CornerDialogConfig } from './CornerDialog.styles';
+import { DefaultButtonProps } from '../Button/Button.props';
+
+export type CornerDialogProps = {
+  custom?: CornerDialogConfig;
+  intent?: 'none' | 'warning' | 'negative';
+  title: string;
+  content: ReactNode;
+  actions?: DefaultButtonProps[];
```

Comment:
There should be add some limitation i guess; end user shouldn't deal with `size` prop, it should be always `small` option. Another unallow prop here is `hasDropdownIndicator`.
```suggestion
  actions?: Omit<DefaultButtonProps, 'size' | 'hasDropdownIndicator'>[];
```

Reply:
Above solution cause typescript error: 
https://github.com/VirtusLab/tetrisly-react/actions/runs/8845324314/job/24288992552?pr=135

I tried to find reason of that behaviour and [I isolated problem to Typescript playground](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAIhBmBDArgG2ABQE4EsC2iWIAyhAMYD2AdgCaEhQC8UA3gFBRSJiSGJVkIAfgBcUAORhcBIuKgAfCQGdy1OrIDcHKDirAIe0RKrUIcxeKXIygpUvMSaEJcCzXgOAG5mtAXy1soJCwCCjoAJJU3lj6NEys2ty8WPyCRuK60bHiWpy6+oZi4iZUPmz+bIHg0BiEHoiocEhowABCyMDA1NgUYErx7JyehDj8wOlOzeg52ko4AF7CRUoEqKgO4ngQNDjIeBuohADmZb5QAGRQABRNYZjS9KSUtPQKIVPAkVnbAJQBQdAYDgXLgAEYdLwQADyeBwwAAPAAVAA0UAA0lAIAAPAo0foAawgIAo8C4VBAAD54ojMTiDHiySBtEIoDC4UjUWiKdoxKVov9qlAALIgADCyBcFDwPT68TZCNqMVGjVCLXanW6WF6SlRACI5otddyAPTGzhQAB6QkqAOFICBIJw4I83nFkulWtlzAdridEO88vhivqKo+6q6VBlOqg+oWECNbFN5qtNsFAA14sHlbc1R0I1GANq6pIQPgCeMAXUTZs4KaqwQAmvERW6uh7tUWS2XBLqq0na9a2E4yIcsNBni4uGIWxK21Gh+RR+PqJPQdP7cDfc7Ia2pfO2BPgFAyGIsw0c+hw5rtfFEFAkxAsFqsAeV0eaKe6tnVZe89evVAoL3maVg2M4-QAO5wgAFlAEqICcUAkrAm5gv60KwsAQA):

Usage of `Omit<T, K>` breaks distribution because Omit iterates through keys and creates new "flat" object.

![image](https://github.com/VirtusLab/tetrisly-react/assets/167776055/c35b3096-14de-4c87-b25b-f933906db1d7)


I found the same problem on stackoverflow: https://stackoverflow.com/questions/57103834/typescript-omit-a-property-from-all-interfaces-in-a-union-but-keep-the-union-s
and here jcals suggested usage of DistributiveOmit<T, K> instead of just Omit<T, K>:

```ts
type DistributiveOmit<T, K extends keyof any> = T extends any
  ? Omit<T, K>
  : never;
```

With DistributiveOmit everything compiles and works correctly. What do you think about above solution? In my opinion we can use it.
![image](https://github.com/VirtusLab/tetrisly-react/assets/167776055/7b97a24f-1021-4ec4-801c-13971b994b3a)